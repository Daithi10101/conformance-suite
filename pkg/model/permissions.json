[{
    "permission": "ReadAccountsBasic",
    "endpoints": ["/accounts", "/accounts/{AccountId}"],
    "default": true
},
{
    "permission": "ReadAccountsDetail",
    "endpoints": ["/accounts", "/accounts/{AccountId}"]
},
{
    "permission": "ReadBalances",
    "endpoints": ["/balances", "/accounts/{AccountId}/balances"]
},
{
    "permission": "ReadBeneficiariesBasic",
    "endpoints": ["/beneficiaries", "/accounts/{AccountId}/beneficiaries"],
    "default": true
},
{
    "permission": "ReadBeneficiariesDetail",
    "endpoints": ["/beneficiaries", "/accounts/{AccountId}/beneficiaries"]
},
{
    "permission": "ReadDirectDebits",
    "endpoints": ["/direct-debits", "/accounts/{AccountId}/direct-debits"]
},
{
    "permission": "ReadStandingOrdersBasic",
    "endpoints": ["/standing-orders", "/accounts/{AccountId}/standing-orders"],
    "default": true
},
{
    "permission": "ReadStandingOrdersDetail",
    "endpoints": ["/standing-orders", "/accounts/{AccountId}/standing-orders"]
},
{
    "permission": "ReadTransactionsBasic",
    "endpoints": ["/transactions", "/accounts/{AccountId}/transactions",
        "/accounts/{AccountId}/statements/{StatementId}/transactions"
    ],
    "required-one-or-more": ["ReadTransactionsCredits", "ReadTransactionsDebits"],
    "optional": ["ReadPAN"],
    "default": true
},
{
    "permission": "ReadTransactionsDetail",
    "endpoints": ["/transactions", "/accounts/{AccountId}/transactions",
        "/accounts/{AccountId}/statements/{StatementId}/transactions"
    ],
    "required-one-or-more": ["ReadTransactionsCredits", "ReadTransactionsDebits"],
    "optional": ["ReadPAN"]
},
{
    "permission": "ReadTransactionsCredits",
    "endpoints": ["/transactions", "/accounts/{AccountId}/transactions",
        "/accounts/{AccountId}/statements/{StatementId}/transactions"
    ],
    "required-one-or-more": ["ReadTransactionsBasic", "ReadTransactionsDetail"],
    "optional": ["ReadPAN"]
},
{
    "permission": "ReadTransactionsDebits",
    "endpoints": ["/transactions", "/accounts/{AccountId}/transactions",
        "/accounts/{AccountId}/statements/{StatementId}/transactions"
    ],
    "required-one-or-more": ["ReadTransactionsBasic", "ReadTransactionsDetail"],
    "optional": ["ReadPAN"]
},
{
    "permission": "ReadStatementsBasic",
    "endpoints": ["/statements", " /accounts/{AccountId}/statements"]
},
{
    "permission": "ReadStatementsDetail",
    "endpoints": ["/statements", " /accounts/{AccountId}/statements",
        "/accounts/{AccountId}/statements/{StatementId}/file"
    ]
},
{
    "permission": "ReadProducts",
    "endpoints": ["/products", "/accounts/{AccountId}/product"]
},
{
    "permission": "ReadOffers",
    "endpoints": ["/offers", "/accounts/{AccountId}/offers"]
},
{
    "permission": "ReadParty",
    "endpoints": ["/accounts/{AccountId}/party"]
},
{
    "permission": "ReadPartyPSU",
    "endpoints": ["/party"]
},
{
    "permission": "ReadScheduledPaymentsBasic",
    "endpoints": ["/scheduled-payments", "/accounts/{AccountId}/scheduled-payments"],
    "default": true
},
{
    "permission": "ReadScheduledPaymentsDetail",
    "endpoints": ["/scheduled-payments", "/accounts/{AccountId}/scheduled-payments"]
},
{
    "permission": "ReadPAN",
    "endpoints": ["", ""]
}
]