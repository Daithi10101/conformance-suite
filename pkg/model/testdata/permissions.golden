[
    {
        "Code": "ReadAccountsBasic",
        "Endpoints": [
            "/accounts",
            "/accounts/{AccountId}"
        ],
        "Default": true,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadAccountsDetail",
        "Endpoints": [
            "/accounts",
            "/accounts/{AccountId}"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadBalances",
        "Endpoints": [
            "/balances",
            "/accounts/{AccountId}/balances"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadBeneficiariesBasic",
        "Endpoints": [
            "/beneficiaries",
            "/accounts/{AccountId}/beneficiaries"
        ],
        "Default": true,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadBeneficiariesDetail",
        "Endpoints": [
            "/beneficiaries",
            "/accounts/{AccountId}/beneficiaries"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadDirectDebits",
        "Endpoints": [
            "/direct-debits",
            "/accounts/{AccountId}/direct-debits"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadStandingOrdersBasic",
        "Endpoints": [
            "/standing-orders",
            "/accounts/{AccountId}/standing-orders"
        ],
        "Default": true,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadStandingOrdersDetail",
        "Endpoints": [
            "/standing-orders",
            "/accounts/{AccountId}/standing-orders"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadTransactionsBasic",
        "Endpoints": [
            "/transactions",
            "/accounts/{AccountId}/transactions",
            "/accounts/{AccountId}/statements/{StatementId}/transactions"
        ],
        "Default": true,
        "RequiredOneOrMore": [
            "ReadTransactionsCredits",
            "ReadTransactionsDebits"
        ],
        "Optional": [
            "ReadPAN"
        ]
    },
    {
        "Code": "ReadTransactionsDetail",
        "Endpoints": [
            "/transactions",
            "/accounts/{AccountId}/transactions",
            "/accounts/{AccountId}/statements/{StatementId}/transactions"
        ],
        "Default": false,
        "RequiredOneOrMore": [
            "ReadTransactionsCredits",
            "ReadTransactionsDebits"
        ],
        "Optional": [
            "ReadPAN"
        ]
    },
    {
        "Code": "ReadTransactionsCredits",
        "Endpoints": [
            "/transactions",
            "/accounts/{AccountId}/transactions",
            "/accounts/{AccountId}/statements/{StatementId}/transactions"
        ],
        "Default": true,
        "RequiredOneOrMore": [
            "ReadTransactionsBasic",
            "ReadTransactionsDetail"
        ],
        "Optional": [
            "ReadPAN"
        ]
    },
    {
        "Code": "ReadTransactionsDebits",
        "Endpoints": [
            "/transactions",
            "/accounts/{AccountId}/transactions",
            "/accounts/{AccountId}/statements/{StatementId}/transactions"
        ],
        "Default": true,
        "RequiredOneOrMore": [
            "ReadTransactionsBasic",
            "ReadTransactionsDetail"
        ],
        "Optional": [
            "ReadPAN"
        ]
    },
    {
        "Code": "ReadStatementsBasic",
        "Endpoints": [
            "/statements",
            "/accounts/{AccountId}/statements"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadStatementsDetail",
        "Endpoints": [
            "/statements",
            "/accounts/{AccountId}/statements",
            "/accounts/{AccountId}/statements/{StatementId}/file"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadProducts",
        "Endpoints": [
            "/products",
            "/accounts/{AccountId}/product"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadOffers",
        "Endpoints": [
            "/offers",
            "/accounts/{AccountId}/offers"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadParty",
        "Endpoints": [
            "/accounts/{AccountId}/party"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadPartyPSU",
        "Endpoints": [
            "/party"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadScheduledPaymentsBasic",
        "Endpoints": [
            "/scheduled-payments",
            "/accounts/{AccountId}/scheduled-payments"
        ],
        "Default": true,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadScheduledPaymentsDetail",
        "Endpoints": [
            "/scheduled-payments",
            "/accounts/{AccountId}/scheduled-payments"
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    },
    {
        "Code": "ReadPAN",
        "Endpoints": [
            "",
            ""
        ],
        "Default": false,
        "RequiredOneOrMore": [],
        "Optional": []
    }
]
